pkg/solana/txm/txm.go

```diff
@@ -84,9 +89,24 @@ func NewTxm(chainID string, tc func() (client.ReaderWriter, error), cfg config.C
 		txs:    newPendingTxContextWithProm(chainID),
 		ks:     ks,
 		client: utils.NewLazyLoad(tc),
```

Comment:
LazyLoad here and in BlockHistoryEstimator forces an RPC to be reused for all read operations until it's reset on transaction error. With `TransactionSneder` we might not observe the error and will keep using unhealthy RPC.
We should replace `*utils.LazyLoad[client.ReaderWriter]` with a corresponding interface and fully inject MultiNode when it's enabled.